source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.h

```diff
@@ -12,6 +12,19 @@ namespace Network {
 
 DECLARE_FACTORY(GetAddrInfoDnsResolverFactory);
 
+// Trace information for getaddrinfo.
+enum class GetAddrInfoTrace : uint8_t {
```

Comment:
I think this is good for tracing which parts of the code were hit before a timeout happened, but I wonder if we should also have actual timing information (like from the NotStarted state to the Started state took N ms, etc).  What do you think @alyssawilk ?